addons/account_asset/account_asset.py

```diff
@@ -357,124 +331,144 @@ def open_entries(self, cr, uid, ids, context=None):
+
 
-class account_asset_history(osv.osv):
+class product_template(models.Model):
+    _inherit = 'product.template'
+    asset_category_id = fields.Many2one('account.asset.category', string='Asset Type', ondelete="restrict")
+    deferred_revenue_category_id = fields.Many2one('account.asset.category', string='Deferred Revenue Type', ondelete="restrict")
+
+
+class account_asset_history(models.Model):
     _name = 'account.asset.history'
     _description = 'Asset history'
-    _columns = {
-        'name': fields.char('History name', select=1),
-        'user_id': fields.many2one('res.users', 'User', required=True),
-        'date': fields.date('Date', required=True),
-        'asset_id': fields.many2one('account.asset.asset', 'Asset', required=True),
-        'method_time': fields.selection([('number','Number of Depreciations'),('end','Ending Date')], 'Time Method', required=True,
-                                  help="The method to use to compute the dates and number of depreciation lines.\n"\
-                                       "Number of Depreciations: Fix the number of depreciation lines and the time between 2 depreciations.\n" \
-                                       "Ending Date: Choose the time between 2 depreciations and the date the depreciations won't go beyond."),
-        'method_number': fields.integer('Number of Depreciations', help="The number of depreciations needed to depreciate your asset"),
-        'method_period': fields.integer('Period Length', help="Time in month between two depreciations"),
-        'method_end': fields.date('Ending date'),
-        'note': fields.text('Note'),
-    }
     _order = 'date desc'
-    _defaults = {
-        'date': lambda *args: time.strftime('%Y-%m-%d'),
-        'user_id': lambda self, cr, uid, ctx: uid
-    }
+
+    name = fields.Char(string='History name', index=True)
+    user_id = fields.Many2one('res.users', string='User', required=True, default=lambda self: self.env.user)
+    date = fields.Date(required=True, default=fields.Date.context_today)
+    asset_id = fields.Many2one('account.asset.asset', string='Asset', required=True)
+    method_time = fields.Selection([('number', 'Number of Depreciations'), ('end', 'Ending Date')], string='Time Method', required=True,
+        help="The method to use to compute the dates and number of depreciation lines.\n"
+             "Number of Depreciations: Fix the number of depreciation lines and the time between 2 depreciations.\n"
+             "Ending Date: Choose the time between 2 depreciations and the date the depreciations won't go beyond.")
+    method_number = fields.Integer(string='Number of Depreciations', help="The number of depreciations needed to depreciate your asset")
+    method_period = fields.Integer(string='Period Length', help="Time in month between two depreciations")
+    method_end = fields.Date(string='Ending date')
+    note = fields.Text()
```

Comment:
this fields are already defined on categories and assets... this is a kind of truplication :)